src/overlays/actors/ovl_En_Part/z_en_part.c

```diff
@@ -21,23 +26,307 @@ const ActorInit En_Part_InitVars = {
+                    pos.y = this->actor.posRot.pos.y + this->actor.shape.unk_08 * this->actor.scale.y +
+                            Math_Rand_CenteredFloat(50.0f);
+                    pos.z = this->actor.posRot.pos.z + Math_Rand_CenteredFloat(60.0f);
+                    velocity.y = Math_Rand_ZeroOne() + 1.0f;
+                    func_80029618(globalCtx, &pos, &velocity, &accel, Math_Rand_S16Offset(80, 100), 0x19, 0, 1);
+                }
+                break;
+            case 5:
+            case 6:
+            case 7:
+            case 8:
+                for (i = 4; i >= 0; i--) {
+                    pos.x = this->actor.posRot.pos.x + Math_Rand_CenteredFloat(25.0f);
+                    pos.y = this->actor.posRot.pos.y + Math_Rand_CenteredFloat(40.0f);
+                    pos.z = this->actor.posRot.pos.z + Math_Rand_CenteredFloat(25.0f);
+                    func_8002A6B8(globalCtx, &pos, &zero, &zero, 0x28, 7, 0xFF, 0xFF, 0xFF, 0xFF, 0, 0, 0xFF, 1, 9, 1);
+                }
+                break;
+        }
+
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->delay--;
+    this->rotation += this->rotationSpeed;
+}
+
+void func_80ACE5B8(EnPart* this, GlobalContext* globalCtx) {
+    this->action = 3;
+}
+
+void func_80ACE5C8(EnPart* this, GlobalContext* globalCtx) {
+    static Vec3f D_80ACF1D4 = { 0.0f, 8.0f, 0.0f };
+    static Vec3f D_80ACF1E0 = { 0.0f, -1.5f, 0.0f };
+
+    Player* player = PLAYER;
+    Vec3f velocity;
+    Vec3f accel;
+    u8 invincibilityTimer;
+
+    this->delay--;
+    if (this->delay == 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    velocity = D_80ACF1D4;
+    accel = D_80ACF1E0;
+
+    if (sqrt(this->actor.xyzDistFromLinkSq) <= 40.0f) {
+        invincibilityTimer = player->invincibilityTimer;
+        if (player->invincibilityTimer <= 0) {
+            if (player->invincibilityTimer <= -40) {
+                player->invincibilityTimer = 0;
+            } else {
+                player->invincibilityTimer = 0;
+                globalCtx->unk_11D58(globalCtx, -8);
+            }
+        }
+        func_8002F71C(globalCtx, this->actor.parent, (650.0f - this->actor.parent->xzDistFromLink) * 0.04f + 4.0f,
+                      this->actor.parent->posRot.rot.y, 8.0f);
+        player->invincibilityTimer = invincibilityTimer;
+        this->delay = 1;
+    }
+
+    func_80033480(globalCtx, &this->actor.posRot.pos, 0.0f, 1, 300, 150, 1);
+    velocity.x = Math_Rand_CenteredFloat(16.0f);
+    func_80029724(globalCtx, &this->actor.posRot.pos, &velocity, &accel, 20,
+                  (s32)((Math_Rand_ZeroOne() * 5.0f + 12.0f) * 2), -1, 10, 0);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_MONBLIN_GNDWAVE - SFX_FLAG);
+}
+
+void func_80ACE7E8(EnPart* this, GlobalContext* globalCtx) {
+    static Vec3f D_80ACF1EC = { 0.0f, 0.0f, 0.0f };
+
+    Vec3f zero = D_80ACF1EC;
+    f32 dist;
```

Comment:
I would rename `dist` to `diffsSum` or something like that because `Math_SmoothScaleMaxMinF` returns the absolute difference between the computed value and the target, and `dist` is summing position differences with a rotation difference